crates/bevy_text/src/text.rs

```diff
@@ -249,3 +254,57 @@ impl Default for TextStyle {
         }
     }
 }
+
+/// Used to configure the default font used by [`TextStyle`] when the font is set to [`FontRef::Default`]
+#[derive(Default, Resource)]
+pub struct DefaultFont {
+    pub path: Option<AssetPath<'static>>,
+}
+
+/// Loads fonts defined by a [`FontRef`]
+/// If the font has already been loaded then it simply converts the [`FontRef::Path`] to a [`FontRef::Handle`]
+/// Otherwise it tries to load the font with the [`AssetServer`]
+pub fn load_font(
+    mut query: Query<&mut Text, Added<Text>>,
+    asset_server: Res<AssetServer>,
+    fonts: Res<Assets<Font>>,
+    default_font: Res<DefaultFont>,
+    mut event: EventWriter<RequestRedraw>,
+) {
+    let mut needs_redraw = false;
+    for mut text in &mut query {
+        for mut section in &mut text.sections {
+            let path = match &section.style.font {
+                FontRef::Default => {
+                    if let Some(path) = default_font.path.as_ref() {
+                        path
+                    } else if let Some((handle, _)) = fonts.iter().next() {
+                        warn!("Default font not set. Using first available font");
+                        section.style.font = FontRef::Handle(fonts.get_handle(handle));
+                        continue;
+                    } else {
+                        panic!("No default font found");
+                    }
+                }
+                FontRef::Path(path) => path,
+                FontRef::Handle(_) => continue,
+            };
+            let handle = match asset_server.get_load_state(path.clone()) {
+                bevy_asset::LoadState::NotLoaded | bevy_asset::LoadState::Unloaded => {
+                    asset_server.load(path.clone())
+                }
+                bevy_asset::LoadState::Loading | bevy_asset::LoadState::Loaded => {
+                    fonts.get_handle(path.clone())
+                }
+                bevy_asset::LoadState::Failed => panic!("Failed to load font {:?}", path),
+            };
+            section.style.font = FontRef::Handle(handle);
```

Comment:
Not a huge fan of this dataflow. A style type should be "declarative". The user sets a value and the underlying system makes it happen under the hood, without changing the users' declared style.

Setting `style.font = FontRef::Default` should hold that default indefinitely, not serve as a placeholder/queued action that gets overwritten in place. Same goes for `FontRef::Path`.

I understand why you're doing this (we need to hold a strong handle somewhere to keep the font alive). But this weird dataflow isn't worth it from my perspective.

Reply:
I agree, I'm just honestly not sure how else I could do that. I'll try to think of something better.